apps/openmw/mwmechanics/stat.hpp

```diff
@@ -235,26 +235,29 @@ namespace MWMechanics
     class AttributeValue
     {
         int mBase;
-        int mModifier;
+        int mFortified;
+        int mModifier; // net effect of Fortified, Drain & Absorb
         float mDamage; // needs to be float to allow continuous damage
 
     public:
-        AttributeValue() : mBase(0), mModifier(0), mDamage(0) {}
+        AttributeValue() : mBase(0), mFortified(0), mModifier(0), mDamage(0) {}
 
         int getModified() const { return std::max(0, mBase - (int) mDamage + mModifier); }
         int getBase() const { return mBase; }
         int getModifier() const {  return mModifier; }
 
         void setBase(int base) { mBase = std::max(0, base); }
-        void setModifier(int mod) { mModifier = mod; }
+        void setModifiers(int fortify, int drain, int absorb);
 
-        void damage(float damage) { mDamage += damage; }
+        void damage(float damage) { mDamage = std::min(mDamage + damage, (float)(mBase + mFortified)); }
```

Comment:
Why only take the fortifier into account? Why not just mDamage = std::min(mDamage + damage, getModified()); ?

I've tested using "drain intelligence on self" and "damage intelligence on self" in vanilla MW and the drain effect is taken into account there, meaning you can't damage your intelligence further if it would cause the modified value to go below zero.